src/sql/workbench/api/common/extHostModelView.ts

```diff
@@ -1370,6 +1370,13 @@ class TextComponentWrapper extends ComponentWrapper implements azdata.TextCompon
 	public set requiredIndicator(requiredIndicator: boolean) {
 		this.setProperty('requiredIndicator', requiredIndicator);
 	}
+
+	public get headingLevel(): number | undefined {
```

Comment:
there are only 6 heading levels, we should restrict it to h1 to h6, right?

Reply:
https://www.w3.org/MarkUp/html3/headings.html

Reply:
There are a bunch of ways that property could be set so I opted just to catch it on the main thread side if it's wrong and clamp it to a valid value. I considered having the type be an enum or something but that seemed like a bit much - generally I expect people to do the right thing and if they don't at least we'll handle it safely.

Reply:
how about using  union type: `1 | 2 | 3 | 4 | 5 | 6`, so that we catch it at compilation time.

Reply:
For some reason I thought numbers couldn't be union types - yeah I like that better.